packages/response-ops/alerts_table/utils/test.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { DeepPartial } from 'utility-types';
+import MaybeMockedDeep = jest.MaybeMockedDeep;
+
+/**
+ * Utility function to create a partial object mock without losing type information
+ *
+ * @example
+ * ```ts
+ * const mockedObject = createPartialObjectMock<MyType>({ prop1: 'value' });
+ * ```
+ */
+export const createPartialObjectMock = <T>(partialObject: DeepPartial<T>) =>
```

Comment:
This allows us to define only a subset of object properties for very large objects without losing type checking on the input partial object and returned value. Should save us a lot of type casts and improve automatic code refactors in IDEs